format/luajit/luajit.go

```diff
@@ -0,0 +1,381 @@
+			d.FieldStruct("ktab", LuaJITDecodeKTabK)
+		}
+	})
+
+	d.FieldArray("khash", func(d *decode.D) {
+		for i := uint64(0); i < nhash; i++ {
+			d.FieldStruct("khash", func(d *decode.D) {
+				d.FieldStruct("k", LuaJITDecodeKTabK)
+				d.FieldStruct("v", LuaJITDecodeKTabK)
+			})
+		}
+	})
+}
+
+func LuaJITDecodeI64(d *decode.D) {
+	d.FieldAnyFn("i64", func(d *decode.D) any {
+		lo := d.ULEB128()
+		hi := d.ULEB128()
+		return int64((hi << 32) + lo)
+	})
+}
+
+func LuaJITDecodeU64(d *decode.D) {
+	d.FieldAnyFn("u64", func(d *decode.D) any {
+		lo := d.ULEB128()
+		hi := d.ULEB128()
+		return (hi << 32) + lo
+	})
+}
+
+func LuaJITDecodeComplex(d *decode.D) {
+	d.FieldAnyFn("real", func(d *decode.D) any {
+		rlo := d.ULEB128()
+		rhi := d.ULEB128()
+		r := (rhi << 32) + rlo
+		return u64tof64(r)
+	})
+
+	d.FieldAnyFn("imag", func(d *decode.D) any {
+		ilo := d.ULEB128()
+		ihi := d.ULEB128()
+		i := (ihi << 32) + ilo
+		return u64tof64(i)
+	})
+}
+
+func LuaJITDecodeKGC(d *decode.D) {
+	kgctype := d.FieldULEB128("type", &kgcType{})
+
+	if kgctype >= 5 {
+		sz := kgctype - 5
+		d.FieldUTF8("str", int(sz))
+	} else {
+		switch kgctype {
+		case 0:
+			//child
+
+		case 1:
+			LuaJITDecodeTab(d)
+
+		case 2:
+			LuaJITDecodeI64(d)
+
+		case 3:
+			LuaJITDecodeU64(d)
+
+		case 4:
+			LuaJITDecodeComplex(d)
+		}
+	}
+}
+
+func LuaJITDecodeKNum(d *decode.D) any {
+	lo := d.ULEB128()
+	if lo&1 == 0 {
+		return lo >> 1
+	} else {
+		hi := d.ULEB128()
+		return u64tof64((hi << 32) + (lo >> 1))
+	}
+}
+
+func LuaJITDecodeProto(di *DumpInfo, d *decode.D) {
+	length := d.FieldULEB128("length")
+
+	d.LimitedFn(8*int64(length), func(d *decode.D) {
+		d.FieldStruct("pdata", func(d *decode.D) {
+			var numuv uint64
+			var numkgc uint64
+			var numkn uint64
+			var numbc uint64
+			var debuglen uint64
+
+			d.FieldStruct("phead", func(d *decode.D) {
+				d.FieldU8("flags")
+				d.FieldU8("numparams")
+				d.FieldU8("framesize")
+				numuv = d.FieldU8("numuv")
+				numkgc = d.FieldULEB128("numkgc")
+				numkn = d.FieldULEB128("numkn")
+				numbc = d.FieldULEB128("numbc")
+
+				debuglen = 0
+				if !di.Strip {
+					debuglen = d.FieldULEB128("debuglen")
+					if debuglen > 0 {
+						d.FieldULEB128("firstline")
+						d.FieldULEB128("numline")
+					}
+				}
+			})
+
+			d.FieldArray("bcins", func(d *decode.D) {
+				for i := uint64(0); i < numbc; i++ {
+					d.FieldStruct("ins", func(d *decode.D) {
+						LuaJITDecodeBCIns(d)
+					})
+				}
+			})
+
+			d.FieldArray("uvdata", func(d *decode.D) {
+				for i := uint64(0); i < numuv; i++ {
+					d.FieldU16("uv")
+				}
+			})
+
+			d.FieldArray("kgc", func(d *decode.D) {
+				for i := uint64(0); i < numkgc; i++ {
+					d.FieldStruct("kgc", LuaJITDecodeKGC)
+				}
+			})
+
+			d.FieldArray("knum", func(d *decode.D) {
+				for i := uint64(0); i < numkn; i++ {
+					d.FieldAnyFn("knum", LuaJITDecodeKNum)
+				}
+			})
+
+			if !di.Strip {
+				d.FieldArray("debug", func(d *decode.D) {
```

Comment:
When i played around with the decoder i noticed that data in debug has some strings, is there any known structure to it or should be threaded as opaque blob? if blob should the array be replaced with `d.FieldRawLen("debug", debuglen*8)`?